presto-docs/src/main/sphinx/admin/properties.rst

```diff
@@ -416,6 +416,40 @@ Task Properties
     to become overloaded due to excessive resource utilization. This can also be specified on
     a per-query basis using the ``task_writer_count`` session property.
 
+Writer Scaling Properties
+-------------------------
+
+By default, the number of writer tasks is static. Enabling writer scaling allows
+Presto to dynamically scale out the number of writer tasks rather than
+allocating a fixed number of tasks. Additional tasks are added when the average
+amount of physical data per writer is above a minimum threshold, but only if the
+query is bottlenecked on writing.
+
+Writer scaling is useful with connectors like Hive that produce one or more
+files per writer -- reducing the number of writers results in a larger average
+file size. However, writer scaling can have a small impact on query wall time
+due to the decreased writer parallelism while the writer count ramps up to match
+the needs of the query.
+
+``scale-writers``
+^^^^^^^^^^^^^^^^^
+
+    * **Type:** ``boolean``
+    * **Default value:** ``false``
+
+    Enable writer scaling. This can be specified on a per-query basis
+    using the ``scale_writers`` session property.
+
+``writer-min-size``
+^^^^^^^^^^^^^^^^^^^
+    * **Type:** ``data size``
+    * **Default value:** ``32MB``
+
+    The minimum amount of data that must be written by a writer task before
+    another writer is eligible to be added. As this is per writer task, not
+    per writer, this value is effectively divided by the number of writer
+    tasks, configured via ``task.writer-count``.
```

Comment:
@electrum : do you think that we should change 
> this value is effectively divided by the number of writer
    tasks, configured via ``task.writer-count``.

to 

> this value is effectively divided by the number of writers in a 
    task, configured via ``task.writer-count``.

Reply:
I think the original is more clear, since “writer tasks” is controlled by the “task writer count” config.

Reply:
Ok. But in the statement above, "writer tasks" corresponds to presto writer tasks (instead of drivers). Am I missing something?

Reply:
I see what you mean. The documentation for writer count says “ The number of concurrent writer threads per worker per query.”

Let me try to rephrase this whole thing to avoid using the term “task”.

Thanks for the feedback. It’s hard to know what others find confusing in documentation.